cmd/kyma/alpha/create/module/module.go

```diff
@@ -95,6 +97,13 @@ func (cmd *command) Run(args []string) error {
 		DefaultCRPath: cmd.opts.DefaultCRPath,
 	}
 
+	setupStep := cmd.NewStep(kustomizeSetup.DefaultNewStepMsg)
+	if err := cmd.SetupKustomize(setupStep); err != nil {
```

Comment:
1. Create the const value of `DefaultNewStepMsg` I don't thing it brings too much value, and the dependency is a bit messy here, here only need to write some output message, why I have to introduce dependency from internal`kustomize` package?
2. Why need to generate this `setupSetup` and pass in to SetupKustomize, instead of directly use `cmd.CurrentStep`? although they are same. I think you can simply use `cmd.NewStep`, `cmd.CurrentStep.Successf`

Reply:
1. This constant is used twice in different commands: `alpha create module` and `alpha deploy`. I personally hate code/constant repetition, so this is my take on removing the duplication. Do you have any other idea how could we share 

2. It was like this before, kustomize setup was a separate step with a distinct name. I didn't want to change that.

Reply:
1. I still think it's not a good approach to introduce this dependencies just because somehow the message is coincidentally same. I mean the message in different process can be different in order to provide more specific information. e.g `Setting up kustomize...` can be `Setting up kustomize during create module` in create module command and `Setting up kustomize during deploy` for deploy command. What I mean is we don't need to worry about sharing such values.

2. You can reference the style just a few line after, all these step messages using `cmd.NewStep`, `cmd.CurrentStep`
directly.
```
cmd.NewStep("Parse and build module...")
// Create base resource defs with module root and its sub-layers
if err := module.Inspect(modDef, cmd.opts.ResourcePaths, cmd.CurrentStep, l); err != nil {
cmd.CurrentStep.Failure()
return err
}
cmd.CurrentStep.Successf("Module built")
```

Reply:
Or you  could introduce a new public function in the ksutomize package which already includes the step, something like
```
func SetupWithDefaultStep(cmd *command, verbose bool) error {
	setupStep := cmd.NewStep("Setting up kustomize...")
	return Setup(setupStep, verbose)
}
```

And then call this method instead of creating two steps with the same message

Reply:
@jeremyharisch This was my initial solution. And it was strictly objected by Borja, who said that new steps should be only created by commands. And indeed, now the "NewStep" is only called in the "cmd" package tree

Reply:
@ruanxin ```[...] The message in different process can be different[...]. What I mean is we don't need to worry about sharing such values.```
In my case I WANT to have the same message. I know it could be different, but I WANT the same.
This means I WANT to use the language features to actually give other developers a strict, clear message about my intentions: I WANT it to be the same.
That's why I require this message not only to be the same value (copied in two places) - I require it to be the same instance of string, so if anyone changes the value, it's changed in two places.
If any other dev in the future decides that the message should be different for some reasons, then the dev will introduce necessary changes. But for now I don't see a reason we should give up and accept constant duplication.

Reply:
ok, but tbh, 
first, I still not get the point what's the reason need these messages to be same so strictly? Maybe you can explain more about it, if this is truly the case, in order to prevent difference, then we need have a test to make sure the message generated from different command should be strictly same.

second, with my own experience and common refactoring understanding, define a const value normally will give developer the impression this value should be always same, in very rear case, people will have the thoughts to change it in later refactoring, that's why I insist for this flexible hint message, there is no needs to define them as const, because they can be changed in any time. I think with this point, @clebs shared same thought, by referencing his comment: `not all commands will want the same message when calling NewStep() in the kustomize setup step.`, by moving this message out of `SetupKustomize` funtion but always use those two default const message surrounded is another way to indicate this command should use same message.

In current code base, we can also find many "duplicate" messages existing, like `"Kyma deployed successfully!"`, `"Deploying Kyma"`, developers should not care about this duplication during writing such hint messages, unless we have solid reason.

Reply:
ok, thanks for the feedback @Tomasz-Smelcerz-SAP , since the above discussion between us is related to how to deal with hint message and it's really not something related to critical process, I suggest we end up here. At least we are not prevent further refactoring to improve the code.

Last but not least, regarding point 2. I think you misunderstood me, the thing I want to test is testing **why** these two message needs to be same, if they are different, what kind of business logic will be break, not testing these two message expect to be same, that's of cause not necessary.

Reply:
Regarding point 5, I think this is a major difference between us (to me, it's definitely not treat as code smell, and there is not necessary to get rid of these so called duplication) and will affect later the direction how we maintain the code, I suggest we create a short discussion meeting next week, including few more opinions. What do you think, can you manage to take care of this discussion? @Tomasz-Smelcerz-SAP